src/solver/svd/svd.hpp

```diff
@@ -0,0 +1,72 @@
+
+#pragma once
+
+#include "numeric/dense_matrix.hpp"
+#include <Eigen/SVD>
+
+namespace neon
+{
+/// svd computes the singular value decomposition of an input matrix and
+/// provides access to the left and right singular vectors and the singular
+/// values.
+class svd
+{
+public:
+    svd(){}; // No idea why this is here
+
+    virtual ~svd() = default;
+
+    /// compute thin SVD decomposition of a matrix A
+    virtual void compute(matrix const& A) = 0;
+
+    /// compute thin SVD decomposition of a matrix A with n being the maximum number of singular vectors
+    virtual void compute(matrix const& A, int const n) = 0;
+
+    /// compute thin SVD decomposition of a matrix A with a tolerance on the singular values
+    /// A singular value will be considered nonzero if its value is strictly greater than
+    /// \f$ \vert singular value \vert \leqslant threshold \times \vert max singular value \vert \f$.
+    virtual void compute(matrix const& A, double const tolerance) = 0;
+
+    /// left singular vectors: columns of the thin U matrix
```

Comment:
You can use `\return left singular vectors (columns of the thin U matrix)` and the code docs will format it better.